src/distributions/mod.rs

```diff
@@ -85,6 +85,20 @@ pub struct Weighted<T> {
     pub item: T,
 }
 
+//implementation of Clone for Weighted
+impl<T> Clone for Weighted<T> where T:Clone{
```

Comment:
Builds and tests using #[derive(Clone)], will update pull request pending a push of the relevant commit from my local repo.